File: evenodd.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(void)
{
 vector<int> numbers={12, 37, 5, 42, 8, 3};
 vector<int> even, odd; // even and odd are empty.

 while (!numbers.empty())
 {
  int nums=numbers.back();
  numbers.pop_back(); // delete the last element of numbers.
  if (nums%2 == 0)
    even.push_back(nums); // append new even number to even
  else
    odd.push_back(nums); // append new odd number to odd
 }

 cout << "even=";
 for (auto i: even)
   cout << i << " ";
 cout << endl;
 
 cout << "odd=";
 for (auto i: odd)
   cout << i << " "; 
 cout << endl;

 return 0;
}

File: evenodd.m
clear all;
numbers=[12, 37, 5, 42, 8, 3];
even=[]; 
odd=[];
% even and odd are empty array

while(length(numbers)>0)
  num=numbers(end);
  numbers(end)=[]; % delete the last element of numbers
  if (mod(num ,2) == 0)
    even=[even, num]; % merge previous even array and new even number
  else
    odd=[odd, num]; % merge previous odd array and new odd number
  end
end

